activerecord/lib/active_record/connection_adapters/abstract/connection_handler.rb

```diff
@@ -103,30 +103,32 @@ def connection_pool_list(role = ActiveRecord::Base.current_role)
 
       def establish_connection(config, owner_name: Base, role: ActiveRecord::Base.current_role, shard: Base.current_shard)
         owner_name = StringConnectionOwner.new(config.to_s) if config.is_a?(Symbol)
-
         pool_config = resolve_pool_config(config, owner_name, role, shard)
         db_config = pool_config.db_config
 
-        # Protects the connection named `ActiveRecord::Base` from being removed
-        # if the user calls `establish_connection :primary`.
-        if owner_to_pool_manager.key?(pool_config.connection_specification_name)
+        # If there is an existing pool with the same values as the pool_config
+        # don't remove the connection. Connections should only be removed if we are
+        # establishing a connection on a class that is already connected to a different
+        # configuration.
+        pool = retrieve_connection_pool(pool_config.connection_specification_name, role: role, shard: shard)
+
+        unless pool && pool.db_config == db_config
```

Comment:
Does this retain the `ActiveRecord::Base` / `:primary` protection mentioned in the above-removed comment? (Assuming we need to)

Reply:
That was for when we were deprecating primary as a name for ActiveRecord::Base and since that's been deprecated and removed I think we no longer need this. I wrote the original comment and went back to the original PR to verify that.